### DISCUSSION TITLE
Question on changes getting overwritten after deployment

### DISCUSSION THREAD

**Discussion by irfanlai** [scolladon/sfdx-git-delta#668] [Q&A]
This is just a question 
---

We use Github actions for Salesforce deployment using sfdx-git-delta. Pipeline is not configured to deploy destructive changes.
Our branching strategy is all feature branches are created from main(Production).
Dev will commit their changes in the branch and create a PR to UATSandbox which is again created from main. 
Once PR is merged the changes gets deployed to UAT Org. Same branch will be used to deploy to main/Prod

**My scenario and question:**
Dev A creates a XYZ branch from main and add some new functions in a common Util class. The changes gets deployed to UAT.
Dev B creates a QQQ branch from main and modify an existing code in the same Util class. The changes gets deployed to UAT and removing Dev A changes

**Actual Result:**
When Dev A branch gets deployed to UAT the changes look good as expected
When Dev B branch gets deployed to UAT then Dev A branch changes gets wiped which jeopardizes the effort of Dev A
This created a huge problem as changes are getting overwritten when same file is getting updated.

**Expected Result:** 
What should have happened in such case? I am assuming both changes should be on UAT.
Open to suggestions.

**Note: What happens if it's a flow or page layout? Will it work in a different way when the same file gets deployed**

**Comment by scolladon** [comment]
Hi @irfanlai 

Thanks for raising this very interesting question.
I convert this into a discussion.

**Comment by scolladon** [comment]
Hi @irfanlai,

I have a few questions, this will help me understand what is going on with git here.

When you say : "Dev create a PR to UATSandbox which is again created from main", you mean there is a UATSandbox branch created from main ? or the branch used to create the PR is created from main ? 

What trigger the deployment to UAT sandbox ? Does it deploys to UAT before the merge of the feature branch to UATSandbox or after ?

When creating the PR to UATSandbox with the QQQ branch, does the code from the XYZ branch is visible in the diff ?

I'll probably have more questions after that 😅 
I tried to reproduce this workflow locally to understand better what is happening.
If you could send a repo isolating exactly what you are currently facing with git it would be much more easier and clearer for me to understand what is happening.

**Reply by irfanlai** [reply]
Let me explain the branching strategy which should answer all your questions.

We have following branches.
main -- Production Org
UATSandbox -- UAT Org (This branch has been created from main)
QASandbox -- QA Org (This branch has been created from main)
We don't have any branch for Dev Org as devs directly creates their changes on Dev Org or locally and push it to Dev Org before promoting them to higher environments.

Feature1 (branch) - Dev will create this branch from main and commit their changes in the branch.
Dev will raise PR for Feature1 to QASandbox which will run validation, once PR is merged to the QASandbox branch then deployment will kick off to QA org. Similarly to move it to UAT org, Dev will raise PR to UATSandbox using the same branch.
For production org deployment. RM will create a Release branch and merge all Feature branches in it and raise PR for main and merge it for deployment to prod.

I have a repo that I shared with you earlier in some other bug. I will update that and share it with you soon,

Feel free to ask any questions if I am still not explaining well.

**Reply by scolladon** [reply]
Is it happening at the repository level ? Or is it ok in the repo, meaning both `Feature1` and `Feature2` code are in `UATSandbox`, but when using `sgd` it somehow erase `Feature1`?

If you could give me access to the repo where I can see the code from `Feature1` deleted in the `UATSandbox` after `Feature2` is merged it would greatly help me understand what happen.
